crates/router_env/src/logger/setup.rs

```diff
@@ -148,14 +142,24 @@ fn setup_tracing_pipeline(
             "service.name",
             service_name,
         )]));
-
-    let tracer = opentelemetry_otlp::new_pipeline()
+    let traces_layer_result = opentelemetry_otlp::new_pipeline()
         .tracing()
         .with_exporter(get_opentelemetry_exporter(config))
         .with_trace_config(trace_config)
-        .install_simple();
-
-    Some(tracer)
+        .install_simple()
+        .map(|tracer| tracing_opentelemetry::layer().with_tracer(tracer));
+
+    if config.ignore_errors {
+        traces_layer_result
+            .map_err(|error| {
+                eprintln!("Failed to create an `opentelemetry_otlp` tracer: {error:?}")
+            })
+            .ok()
+    } else {
+        // Safety: This is conditional, there is an option to avoid this behavior at runtime.
+        #[allow(clippy::expect_used)]
+        Some(traces_layer_result.expect("Failed to create an `opentelemetry_otlp` tracer"))
```

Comment:
```suggestion
        traces_layer_result.expect("Failed to create an `opentelemetry_otlp` tracer")
```

Reply:
I'll need the `Some` since this allows us to either provide a trace layer when traces are enabled or `None` when traces are disabled. This works out since `Option<_>` also implements the required traits.